src/zproto_codec_c_v1.gsl

```diff
@@ -1225,6 +1229,136 @@ $(class.name)_print ($(class.name)_t *self)
     }
 }
 
+// create new array of unsigned char from properly encoded string
+// len of resulting array is strlen (str) / 2
+// caller is responsibe for freeing up the memory
+static byte*
+s_bytes_new (const char* str)
+{
+    if (!str || (strlen (str) % 2) != 0)
+        return NULL;
+
+    size_t strlen_2 = strlen (str) / 2;
+    byte *mem = (byte*) zmalloc (strlen_2);
+
+    for (size_t i = 0; i != strlen_2; i++)
+    {
+        char buff[3] = {0x0, 0x0, 0x0};
+        strncpy (buff, str, 2);
+
+        byte b;
+        sscanf (buff, "%x" SCNu8, &b);
+
+        mem [i] = b;
+        str += 2;
+    }
+
+    return mem;
+}
+
+// convert len bytes to hex string
+// caller is responsibe for freeing up the memory
+static char *
+s_bytes_str (byte* mem, size_t len)
+{
+    assert (mem);
+    char* ret = zmalloc (4*len);
+    char* aux = ret;
+    for (size_t i = 0; i != len; i++)
+    {
+        if (mem [i] <= 0xf)
+            sprintf (aux, "0%x", mem [i]);
+        else
+            sprintf (aux, "%x", mem [i]);
```

Comment:
Could this be simplified by using just `sprintf (aux, "%02x", mem [i]);` ?
Then we do not need the if-else-clause.